webui/src/pages/extension-detail/extension-detail-overview.tsx

```diff
@@ -163,11 +163,47 @@ export const ExtensionDetailOverview: FunctionComponent<ExtensionDetailOverviewP
         </>;
     };
 
-    const renderWorksWithList = (downloads: { [targetPlatform: string]: UrlString }): ReactNode => {
-        return Object.keys(downloads).map((targetPlatform, index) => {
-            const displayName = getTargetPlatformDisplayName(targetPlatform);
-            return displayName ? <span key={targetPlatform}>{index > 0 ? ', ' : ''}{displayName}</span> : null;
-        });
+    const renderWorksWithEngines = (engines?: Record<string, string>): ReactNode => {
```

Comment:
I'd be great to make this use `useCallback` so that we memoize the return value without having to re-render every time. The same goes for `renderWorksWithTargetPlatformsList` as well.

Reply:
If we do, we should make the functions depend on `extension.engines` and `extension.downloads` respectively by removing the arguments and using `extension` there and always reference `extension` from the component itself.

Reply:
Sounds good. I'll add memoization.